crates/oxc_linter/src/rules/eslint/new_cap.rs

```diff
@@ -0,0 +1,795 @@
+use crate::{context::LintContext, rule::Rule, AstNode};
+use oxc_ast::{
+    ast::{ChainElement, ComputedMemberExpression, Expression},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{CompactStr, GetSpan, Span};
+use regex::Regex;
+
+fn new_cap_diagnostic(span: Span, cap: &GetCapResult) -> OxcDiagnostic {
+    let msg = if *cap == GetCapResult::Lower {
+        "A constructor name should not start with a lowercase letter."
+    } else {
+        "A function with a name starting with an uppercase letter should only be used as a constructor."
+    };
+
+    OxcDiagnostic::warn(msg).with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NewCap(Box<NewCapConfig>);
+
+#[derive(Debug, Default, Clone)]
+pub struct NewCapConfig {
+    new_is_cap: bool,
+    cap_is_new: bool,
+    new_is_cap_exceptions: Vec<CompactStr>,
+    new_is_cap_exception_pattern: Option<Regex>,
+    cap_is_new_exceptions: Vec<CompactStr>,
+    cap_is_new_exception_pattern: Option<Regex>,
+    properties: bool,
+}
+
+impl std::ops::Deref for NewCap {
+    type Target = NewCapConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+fn bool_serde_value(map: &serde_json::Map<String, serde_json::Value>, key: &str) -> bool {
+    let Some(value) = map.get(key) else {
+        return true; // default value
+    };
+
+    let err = format!("eslint/new-cap: expect configuration option '{key}' to be a boolean.");
+
+    value.as_bool().expect(&err)
```

Comment:
we typically `unwrap_or_default` in these cases to vaoid panics.

Reply:
I found some rules which are already doing this :)

Removed it with https://github.com/oxc-project/oxc/pull/8146/commits/b8a83e517b5d6c848555bbf05aea830cee5c77e3

Reply:
😬